OpenRA.Mods.Common/ItchIntegration.cs

```diff
@@ -0,0 +1,90 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using OpenRA.Support;
+
+namespace OpenRA.Mods.Common
+{
+	public class ItchIntegration : IGlobalModData
+	{
+		class User
+		{
+			[JsonProperty("url")]
+			public string Url { get; set; }
+
+			[JsonProperty("gamer")]
+			public bool Gamer { get; set; }
+
+			[JsonProperty("id")]
+			public int Id { get; set; }
+
+			[JsonProperty("press_user")]
+			public bool PressUser { get; set; }
+
+			[JsonProperty("developer")]
+			public bool Developer { get; set; }
+
+			[JsonProperty("username")]
+			public string Username { get; set; }
+
+			[JsonProperty("display_name")]
+			public string DisplayName { get; set; }
+		}
+
+		class Root
+		{
+			public User User { get; set; }
+		}
+
+		public void GetPlayerName(Action<string> callback)
+		{
+			Task.Run(async () =>
+			{
+				User user = null;
+
+				try
+				{
+					var apiKey = Environment.GetEnvironmentVariable("ITCHIO_API_KEY", EnvironmentVariableTarget.Process);
+					if (!string.IsNullOrEmpty(apiKey))
+					{
+						var client = HttpClientFactory.Create();
+						client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", apiKey);
+						var httpResponseMessage = await client.GetAsync("https://itch.io/api/1/jwt/me");
+						httpResponseMessage.EnsureSuccessStatusCode();
+						var result = await httpResponseMessage.Content.ReadAsStringAsync();
+						user = JsonConvert.DeserializeObject<Root>(result).User;
+					}
+				}
+				catch (Exception e)
+				{
+					Log.Write("debug", e);
+				}
+
+				var name = "";
+				if (user != null)
+				{
+					if (string.IsNullOrEmpty(user.DisplayName))
+						name = user.Username;
+					else
+						name = user.DisplayName;
+				}
+				else
+					Log.Write("debug", "Failed to query player name from itch.io API.");
```

Comment:
I kinda feel like this will generate redundant logs for third party mods, or even when players download from other sources

Reply:
No, that is not true. When the API key is not set by the itch client then it never gets this far.

Reply:
I see. That was not intentional.